spec/solver_spec.rb

```
@@ -0,0 +1,111 @@
+require_relative '../solver'
+
+describe Solver do
+  it 'should return 1 for 0' do
+    expect(Solver.factorial(0)).to eq(1)
+  end
+  it 'should return 1 for 1' do
+    expect(Solver.factorial(1)).to eq(1)
+  end
+  it 'should return 2 for 2' do
+    expect(Solver.factorial(2)).to eq(2)
+  end
+  it 'should return 6 for 3' do
+    expect(Solver.factorial(3)).to eq(6)
+  end
+  it 'should return 24 for 4' do
+    expect(Solver.factorial(4)).to eq(24)
+  end
+  it 'should return 120 for 5' do
+    expect(Solver.factorial(5)).to eq(120)
+  end
+  it 'should return 1 for -1' do
+    expect(Solver.factorial(-1)).to eq(1)
+  end
+  it 'should return 1 for -2' do
+    expect(Solver.factorial(-2)).to eq(1)
+  end
+  it 'should return 1 for -3' do
+    expect(Solver.factorial(-3)).to eq(1)
+  end
```

- According to the reviewer's suggestion, kindly make sure that when a negative integer is given, it should raise an exception 😉. For example:
```
     describe Solver do
         describe "#factorial" do
             it "raises exception for negative number" do
             solver = Solver.new
             expect { solver.factorial(-5) }.to raise_exception
             end
         end
     end
```